server/consumer.go

```diff
@@ -1586,6 +1586,27 @@ func (o *consumer) processAckMsg(sseq, dseq, dc uint64, doSample bool) {
 	}
 }
 
+// Determine if this is a truly filtered consumer. Modern clients will place filtered subjects
+// even if the stream only has a single non-wildcard subject designation.
+// Read lock should be held.
+func (o *consumer) isFiltered() bool {
+	if o.cfg.FilterSubject == _EMPTY_ {
+		return false
+	}
+	// If we are here we want to check if the filtered subject is
+	// a direct match for our only listed subject.
+	mset := o.mset
+	if mset == nil {
+		return true
+	}
+	cfg := mset.cfg
```

Comment:
cfg is a struct, so that makes it a copy, which can be expensive since you call that for every ack'ed message. If you just don't want to repeat mset.cfg, at least do: `cfg := &mset.cfg`

Reply:
Thanks.